### PR TITLE
Create global test environment

### PR DESCRIPTION
**Pull request by mmayadag**
- Create global test environment. 
- Add contributors sections to readme.
- Create separate test file.
- Update packages.

### CODE REVIEW

File: setup/puppeteer_environment.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      throw new Error('wsEndpoint not found');
+    }
+
+    // connect to puppeteer
+    this.global.__BROWSER__ = await puppeteer.connect({
+      browserWSEndpoint: wsEndpoint,
+    });
+  }
+
+  async teardown() {
```
Comment (ilhan-mstf) [inline setup/puppeteer_environment.js:30]: There are teardown function and teardown.js file. What is the difference?

File: setup/puppeteer_environment.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+const puppeteer = require('puppeteer');
+const os = require('os');
+const DIR = path.join(os.tmpdir(), 'jest_puppeteer_global_setup');
+
+class PuppeteerEnvironment extends NodeEnvironment {
+  constructor(config) {
+    super(config);
+  }
+
+  async setup() {
```
Comment (ilhan-mstf) [inline setup/puppeteer_environment.js:15]: There are setup function and setup.js file. What is the difference?

Comment (mmayadag) [reply setup/puppeteer_environment.js:15]: Setup.js is **GlobalSetup** script
Pupeteer_environment is a puppeteer for **each Test Environment**

File: setup/puppeteer_environment.js (hunk 1 above)
Comment (mmayadag) [reply setup/puppeteer_environment.js:30]: We can close the **puppeteer instance** and clean-up the file.
This is only for one custom puppeteer instance's teardown.

For example imagine 1 window chrome browser is Global Browser < which is created with setup.js
  - Every tab is new page for puppeteer created from puppeteer_environment.js .
  - firstly we are init global setup than each test puppeteer .
  - than closing it with this teardown function.
  - Finally after complete each test , We are closing global browser instance with teardown.js 

For example , We can create different puppeteer instance for each different test file.
**But teardown.js is closing global browser.**

File: setup/puppeteer_environment.js (hunk 2 above)
Comment (mmayadag) [reply setup/puppeteer_environment.js:15]: https://github.com/ilhan-mstf/jest-puppeteer-test-automation/pull/1#discussion_r335477058

**Review by ilhan-mstf — CHANGES_REQUESTED** [review]
- Can you remove `package-lock.json` and add it to the `.gitignore`?
- Can you rename the `test1.js` file and also the folder `__tests__`? What is the logic behind `__tests__` naming convention?

File: setup/teardown.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+const rimraf = require('rimraf');
+const path = require('path');
+
+const DIR = path.join(os.tmpdir(), 'jest_puppeteer_global_setup');
+module.exports = async function() {
+  // close the browser instance
+  print('Teardown Puppeteer','green');
+  await global.__BROWSER_GLOBAL__.close();
+
+  // clean-up the wsEndpoint file
```
Comment (ilhan-mstf) [inline setup/teardown.js:13]: What is wsEndpoint?

**Comment by mmayadag** [comment]
> * package-lock.json

https://acdcjunior.github.io/npm-package-locks_npm_notice_created_a_lockfile_as_package-lock-json_you_should_commit_this_file.html

Should I commit it?
**Yes.**

> package-lock.json is automatically generated for any operations where npm modifies either the node_modules tree, or package.json. It describes the exact tree that was generated, such that subsequent installs are able to generate identical trees, regardless of intermediate dependency updates.

**I should ask again  really want to remove that?** 
https://github.com/ilhan-mstf/jest-puppeteer-test-automation/pull/1#pullrequestreview-302769407

File: setup/teardown.js (hunk 3 above)
Comment (mmayadag) [reply setup/teardown.js:13]: **browserWSEndpoint** <?string> a browser websocket endpoint to connect to.
A browser url to connect to, in format http://${host}:${port}. Use interchangeably with browserWSEndpoint to let Puppeteer fetch it from metadata endpoint.

> // clean-up the wsEndpoint file
Basicly its end instance.

**Comment by ilhan-mstf** [comment]
We don't need package-lock.json in this repo. Npm needs it, and automatically creates when it is absent.

**Comment by mmayadag** [comment]
https://github.com/ilhan-mstf/jest-puppeteer-test-automation/pull/1#issuecomment-544924803
package-lock.json removed

**Review by ilhan-mstf — APPROVED** [review]

**Pull request merged**